### PR TITLE
Allow configuring max retries

### LINKED ISSUE #35: Allow configuring max retries
Provide a way to externally configure the maximum amount of retries handlers get executed before declaring the message as poison and rejecting it

### PR DESCRIPTION
**Pull request by avigailberger**
retries

Provide a way to externally configure the maximum amount of retries handlers get executed before declaring the message as poison and rejecting it

Closes #35

### CODE REVIEW

File: gbus/abstractions.go [hunk 1]
```diff
@@ -166,6 +166,9 @@ type Builder interface {
 	//ConfigureHealthCheck defines the default timeout in seconds for the db ping check
 	ConfigureHealthCheck(timeoutInSeconds time.Duration) Builder
 
+	//RetriesNum defines the number of retries upon error
+	RetriesNum(retries uint) Builder
```
Comment (rhinof) [inline gbus/abstractions.go]: As there are going to be more configuration options that are currently hard coded but moving forward we would like to provide a way to configure I think we would probably be better off have one WithConfiguration(ConfigObject) that we can use to place on it additional config options moving forward.
WDYT ?

File: gbus/builder/builder.go [hunk 2]
```diff
@@ -170,6 +170,11 @@ func (builder *defaultBuilder) ConfigureHealthCheck(timeoutInSeconds time.Durati
 	return builder
 }
 
+func (builder *defaultBuilder) RetriesNum(retries uint) gbus.Builder {
+	gbus.MaxRetryCount = retries
```
Comment (rhinof) [inline gbus/builder/builder.go]: should guard against zero and negative values

Comment (vladshub) [inline gbus/builder/builder.go]:
```suggestion
        if retries > 0 {
        	gbus.MaxRetryCount = retries
        }
```

File: gbus/worker.go [hunk 3]
```diff
@@ -391,10 +393,15 @@ func (worker *worker) invokeHandlers(sctx context.Context, handlers []MessageHan
...
 
-	//retry for MaxRetryCount, back off by a Fibonacci series 50, 50, 100, 150, 250 ms
+	//retry for MaxRetryCount, back off by a jittered stratergy
+	seed := time.Now().UnixNano()
+	random := rand.New(rand.NewSource(seed))
 	return retry.Retry(action,
 		strategy.Limit(MaxRetryCount),
-		strategy.Backoff(backoff.Fibonacci(50*time.Millisecond)))
+		strategy.BackoffWithJitter(
+			backoff.BinaryExponential(10*time.Millisecond),
```
Comment (rhinof) [inline gbus/worker.go]: lets have these values exposed via configuration with default values

Comment (avigailberger) [reply gbus/worker.go]: just the `10*time.Millisecond`, right?

File: gbus/abstractions.go [hunk 4]
```diff
@@ -15,6 +15,12 @@ const (
 	EVT Semantics = "evt"
 )
 
+//ConfigObject provides configuration passed to the bus builder
+type ConfigObject struct {
```
Comment (rhinof) [inline gbus/abstractions.go]: Please rename to BusConfiguration

**Pull request merged**